olive/passes/onnx/optimum_conversion.py

```diff
@@ -70,42 +76,64 @@ def _run_for_config(
-        if len(hf_component_names) == 0:
-            return ONNXModelHandler(resolve_onnx_path(output_model_path))
-        elif len(hf_component_names) == 1:
-            return ONNXModelHandler(Path(output_model_path) / hf_component_names[0])
-
-        model_components = []
-        model_component_names = []
-        for component_name in hf_component_names:
-            model_components.append(
-                ONNXModelHandler(str(Path(output_model_path) / component_name), model_attributes=model.model_attributes)
+        if version.parse(optimum_version.__version__) < version.parse("1.14.0"):
+            logger.warning(
+                "The behavior of Optimum onnx exporter changed in version 1.14.0 with the introduction of `legacy`"
+                " option. You are using an older version of optimum so it will use the legacy behavior and the output"
+                " model/s may not be the same as the latest version. Please upgrade to the latest version of optimum if"
+                " you do not want the legacy behavior!"
             )
-            model_component_names.append(component_name)
+            if "legacy" in extra_args:
+                logger.warning(
+                    "`legacy` option is set in the extra_args, but it is ignored because you are using optimum<1.14.0."
+                )
+                del extra_args["legacy"]
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            export_optimum_model(
+                model.model_path or hf_config.model_name,
+                temp_dir,
+                **extra_args,
+            )
+
+            exported_models = [name.stem for name in Path(temp_dir).iterdir() if name.suffix == ".onnx"]
+            logger.debug(f"Exported models: {exported_models}")
+            if config["components"]:
+                assert all(
+                    component in exported_models for component in config["components"]
+                ), f"Components {config['components']} are not exported. Only {exported_models} are exported."
+            components = config["components"] or exported_models
+            logger.debug(f"Components to export: {components}")
+
+            # if there is only one component, return it directly
+            if len(components) == 1:
+                output_model_path = resolve_onnx_path(output_model_path)
+                model_proto = onnx.load(Path(temp_dir) / f"{components[0]}.onnx")
+                return model_proto_to_olive_model(model_proto, output_model_path, config)
+
+            # if there are multiple components, return a composite model
+            model_components = []
+            model_component_names = []
+            for component_name in components:
+                # save the component model to a sub-directory
+                component_output_path = resolve_onnx_path(Path(output_model_path).with_suffix("") / component_name)
+                component_proto = onnx.load(Path(temp_dir) / f"{component_name}.onnx")
```

Comment:
>onnx.load(Path(temp_dir) / f"{component_name}.onnx") [](http://example.com/codeflow?start=34&length=52)

if the model is large(in mistral case, it is 28GB), then the load/resave will be time consuming.

Reply:
removed